homeassistant/components/bmw_connected_drive/config_flow.py

```diff
@@ -115,10 +115,6 @@ async def async_step_account_options(
                         CONF_READ_ONLY,
                         default=self.config_entry.options.get(CONF_READ_ONLY, False),
                     ): bool,
-                    vol.Optional(
-                        CONF_USE_LOCATION,
```

Comment:
Please remove the corresponding string from strings.json too.

Reply:
Removing this option could be considered a breaking change.

Reply:
I don't think this is a breaking change, as we have to set `CONF_USE_LOCATION` to `True`.
That is a limitation of the new BMW API, which also requires you to allow using your phone's location to be able to see your vehicle's location. 

However, when thinking about it... We could consider it a breaking change for privacy reasons. In this case, I would opt for keeping the config flow option in and throwing errors/not executing the `find_vehicle` service at all if the option is not set.

Reply:
I don't think we need the option, but we should make it clear that this has changed by adding a breaking change paragraph in the PR description.

If the user hasn't allowed the location access in the phone, we can raise an error when calling the find vehicle service.

Reply:
Activating location access in the BMW app will not enable it in HA.
As far as I understand the API, you either send a location with the HTTP request for `find_vehicle` and get the location of your vehicle as response, or the vehicle might update some thing in the BMW API but you do not get a direct vehicle location (it might be updated into general status on some cars though).

It is also fine (for more recent cars) if you send a dummy location like the HA default location or (0.0, 0.0) as long as the attributes are given to the API.

Reply:
What's the use case for reporting your location to BMW servers? It seems it's only interesting to know the location of the vehicle.

Reply:
The problem is BMW wanting to know where you are (whyever they do that). Without sending your `latitude` and `longitude` in the request headers to BMW's servers, the API will respond with `HTTP 400` and the following error message (both have to exist): 
```json
{
    "statusCode": 400,
    "error": "Bad Request",
    "message": "Latitude is required"
}
```

However this is only required if one is actively asking BMW's server, where the vehicle is located. Newer vehicle (at least after 7/2014) also report their location to BMW's server automatically when shutting down the engine. 

If this happens, the `vehicle_finder` service does not need to be used at all as for these vehicles the location data is available in the default status returned by the API.
Older vehicles have to call this service as they will otherwise not get any location data at all.

Reply:
So is the user locations m only needed when the user uses the `find_vehicle` service? If so, it's enough that we announce this in the breaking change paragraph. Then the users can choose whether they want to use the service or not.

Reply:
Exactly. I have updated the breaking changes paragraph of this PR and also stated it in the docs PR (linked in breaking changes).

Reply:
Good!

Reply:
Please just remove the stale string from strings.json and we're good here.